code/datums/materials/basemats.dm

```diff
@@ -161,7 +161,8 @@ Unless you know what you're doing, only use the first three numbers. They're in
 
 /datum/material/bananium/on_applied(atom/source, amount, material_flags)
 	. = ..()
-	source.AddComponent(/datum/component/squeak, list('sound/items/bikehorn.ogg'=1), 50, falloff_exponent = 20)
+	if(!HAS_TRAIT(source, SQUEAKY_COMPONENT_TRAIT))
+		source.AddComponent(/datum/component/squeak, list('sound/items/bikehorn.ogg'=1), 50, falloff_exponent = 20)
```

Comment:
Use LoadComponent instead.

Reply:
As in, you can remove this trait entirely and just use LoadComponent, which will only add the component if it does not exist.

Reply:
gone